src/main/java/net/quedex/api/common/WebsocketStream.java

```diff
@@ -59,7 +60,12 @@ public void onClose(int code, String reason, boolean remote) {
 
             @Override
             public void onError(Exception ex) {
-                WebsocketStream.this.onError(new CommunicationException("Websocket error", ex));
+                if (ex instanceof IOException && "Broken pipe".equals(ex.getMessage())) {
```

Comment:
Will it work on Windows and Mac OS?

Reply:
Good point - quick research suggests that it won't. Maybe we should treat all `IOExceptions` as `DisconnectedException`?

Reply:
I think we should ensure that socket is closed before this method receives `IOException`.

Reply:
If be "ensure" you mean:
1. Enforcing this programmatically - this is impossible, we don't have control over the state of the socket and calls of this method.
2. Verifying this in the implementation of the WebSocket library - I've investigated all the places that call this method and it seems that it's used only if a connection is closed or is about to be closed (the next code line following an `onError` closes the connection).

Reply:
2 is enough (we could implement it ourselves, but this is an implementation detail actually).